src/case_gather/xml_parser.py

```diff
@@ -1,181 +1,138 @@
-
-def case_collector(parliament_session_number):
-    xml_logger = logging.getLogger('xmlParser')
-
-    link = "http://www.althingi.is/altext/xml/thingmalalisti/?lthing="
-    soup = get_xml(link + str(parliament_session_number))
+    return soup.find_all('mál')
 
-    xml_logger.info('have the case soup')
+def getCaseNumberFromCase(soup):
+    """
+        Takes in a soup with <mál><mál> as the
+        root tag and returns the málsnumer attribute
+    """
+    return soup["málsnúmer"]
 
-    case_numbers_gen = xml_helper.get_attribute_value(
-        soup, "mál", "málsnúmer")
+def getCaseName(soup):
+    """
+        Takes in a soup and finds the case name
+    """
+    return soup.find("málsheiti").string
 
-    xml_logger.info('have the case numbers generator')
+def getAlthingiLink(soup):
+    """
+        Takes in a soup and returns the althingi link under the html tag
+    """
+    return soup.find("html").string
 
-    case_names_gen = xml_helper.get_element_text(soup, "málsheiti")
+def getCaseType(soup):
+    """
+        Takes in a soup and finds the name of the case type
+    """
+    return soup.find("málstegund").find('heiti').string
 
-    xml_logger.info('have the case names generator')
+def getStatus(soup):
+    """
+        Takes in a soup and returns the status of the case
+    """
+    status = soup.find("staðamáls")
+    if status:
+        return status.string
+    return ""
 
-    case_types_gen = xml_helper.get_attribute_value(
-        soup, "málstegund", "málstegund")
 
-    xml_logger.info('have the case case_types_gen generator')
+def getSubjectNames(soup):
+    """
+        Takes in a soup, finds all subjects and returns
+        a list of their names
+    """
+    subject_names = []
+    for subject_element in soup.find_all('efnisflokkur'):
+        subject_name = subject_element.find("heiti").string
+        subject_names.append(subject_name)
+    return subject_names
 
-    for number, name, case_type in zip(case_numbers_gen, case_names_gen, case_types_gen):
-        yield number, name, case_type
+def getRelatedCases(soup):
+    """
+        Takes in a soup and finds all relevant cases and returns
+        their ids
+    """
+    related_cases = soupUtils.get_attribute_value(soup, "mál", "málsnúmer")
 
+    # remove the first rel_case, as it is the actual case
+    if related_cases:
+        related_cases.pop(0)
+    return related_cases
 
-def get_xml(link):
-    xml_logger = logging.getLogger('xmlParser')
-    try:
-        result = requests.get(link)
-    except Exception as e:
-        xml_logger.error(e.message)
-    else:
-        content = result.content
-        soup = BeautifulSoup(content, features="xml")
-        return soup
-    finally:
-        pass
+def getCaseCreatorNames(soup):
+    """
+        Takes in a soup and finds the 'framsögumaður'
+        goes through them all, adds their names to a list
+        and returns said list
+    """
+    case_creators_elements = soup.find_all("framsögumaður")
```

Comment:
Its just a element in the XML from althingi (search for it [here](http://www.althingi.is/altext/xml/thingmalalisti/thingmal/?lthing=146&malnr=413))

As far as I understand, it means whoever presents the case on althingi (we save it as case creator)